addons/web/static/tests/views/kanban/kanban_view_tests.js

```diff
@@ -5080,6 +5083,7 @@ QUnit.module("Views", (hooks) => {
                 '<field name="foo"/>' +
                 '<field name="product_id" readonly="0" invisible="1"/>' +
                 '<field name="state" readonly="1"/>' +
+                '<field name="currency_id" readonly="True" invisible="1"/>' +
```

Comment:
From what I understand, your changes are specifically to take `readonly="1"` into account, so that's what you should be testing here, and not `readonly="True"`, right?

Reply:
Good catch. @vivp-odoo when you fix a bug, the test you introduce must fail without the fix (for the particular issue you're fixing) and pass with the fix.

For this PR, I would add a new test instead of modifying an existing one (which tests another relevant scenario).

Reply:
There might be something somewhere in the mockserver, but your fix does more than handling "True" and "1", it also handle any kind of python expression, like "partner_id == False". That's something I would test.

Reply:
> There might be something somewhere in the mockserver, but your fix does more than handling "True" and "1", it also handle any kind of python expression, like "partner_id == False". That's something I would test.

Hello @aab-odoo 
My fix does not evaluate any kind of Python expression, as I understand that the existing code requires a specific context for Python expressions in the `evaluateBooleanExpr` function.

Also regarding the drag-and-drop functionality for records grouped by a field, how can we handle any Python expression especially when the values can vary for each record?.

Reply:
Indeed, and as a matter of fact, it crashes if readonly is a python expression depending on some fields (e.g. `readonly="id > 4"` in the arch). But the expression can also be something like `context.get("abc")` which we must evaluate.

This is a bit more complicated than initially expected. I suggest that we take this issue back in our pipe and try to find a proper fix.

Reply:
Hello @awa-odoo , @aab-odoo 

I agree that addressing this issue requires a more comprehensive fix.
we should wait for the resolution of task-3611116.

Thanks !